### ISSUE TITLE
reorder shader editor header to match the script editor's

### ISSUE THREAD

**Issue by mieldepoche** [godotengine/godot-proposals#7702]
### Describe the project you are working on

stuff

### Describe the problem or limitation you are having in your project

Script editor:
![image](https://github.com/godotengine/godot-proposals/assets/71937306/6bcd1782-f922-4725-b220-e46e7b145b77)

Shader editor:
![image](https://github.com/godotengine/godot-proposals/assets/71937306/363f6e59-efba-4dc8-9a35-1bf190321338)

they mismatch so badly it hurts :(

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I think the most established experience is the script editor (everyone uses it) so it should be a baseline.
So: reorder the layout to match the script editor:
- `File` then `Edit` then `Search` then `Go To`, padded to the left
- remove the `Help` button and put an `Online Docs` button to the right
- move the "make floating" button to the far right
- one day if shader docs get integrated to the engine docs (it would be so nice), add that search help button

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

basically like the script editor's header 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

used all day long by some people

### Is there a reason why this should be core and not an add-on in the asset library?

core ux

**Comment by jsjtxietian** [comment]
I suppose that  the origin layout came from the fact that visual shader will reuse this left panel too, and visual shader node don't need functions like `search` or `go to`.

![image](https://github.com/godotengine/godot-proposals/assets/8315986/2c7a9836-59e1-4021-8532-475d8a485020)
